ui/ducks/send/send.js

```diff
@@ -1006,13 +1014,21 @@ const slice = createSlice({
           recipient.error = null;
         }
 
-        if (
-          isSendingToken &&
-          isValidHexAddress(recipient.userInput) &&
-          (toChecksumAddress(recipient.userInput) in contractMap ||
-            checkExistingAddresses(recipient.userInput, tokens))
-        ) {
-          recipient.warning = KNOWN_RECIPIENT_ADDRESS_WARNING;
+        if (isSendingToken && isValidHexAddress(recipient.userInput)) {
+          // When useTokenDetection flag is true the tokenList contains tokens with non-checksum address from the dynamic token service api,
+          // When useTokenDetection flag is false the tokenList contains tokens with checksum addresses from contract-metadata.
+          // So the flag indicates whether the address of tokens currently on the tokenList is checksum or not.
+          const userInput = useTokenDetection
+            ? recipient.userInput
+            : toChecksumAddress(recipient.userInput);
+          if (
+            tokenAddressList.includes(userInput) ||
+            checkExistingAddresses(userInput, tokens)
+          ) {
+            recipient.warning = KNOWN_RECIPIENT_ADDRESS_WARNING;
+          } else {
+            recipient.warning = null;
```

Comment:
Can we initialize `recipient.warning` to `null`?  If so, we can remove this `else` clause and the one underneath it.

Reply:
This is likely fine, double checked to make sure it wouldn't interfere with other validation.